### PR TITLE
Unit tests try catch

### PR DESCRIPTION
**Pull request by Jim8y**
This pr increase coverage of try-catch.

### CODE REVIEW

File: src/Neo.Compiler.CSharp/CompilationEngine.cs [hunk 1]
```diff
@@ -164,6 +164,7 @@ private List<CompilationContext> CompileProjectContracts(Compilation compilation
             var sortedClasses = TopologicalSort(classDependencies);
             Parallel.ForEach(sortedClasses, c =>
             {
+                if (!c.Name.Contains("Contract_TryCatch")) return;
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/CompilationEngine.cs:167]: This should be avoided

Comment (shargon) [inline src/Neo.Compiler.CSharp/CompilationEngine.cs:167]:
```suggestion
```

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:167]: This one is still working. https://github.com/neo-project/neo-devpack-dotnet/pull/1076

**Comment by Jim8y** [comment]
Contract_TryCatch [0x37bd3862fb000b816635be9252943d8d03b0c63c] [73.42 % - 50.62 %]
┌-────────────────────────────────────────────────────────────────────────────────────────────────────────────────-┬-────────-┬-────────-┐
│ Method                                                                                                           │   Line   │   Branch │
├-────────────────────────────────────────────────────────────────────────────────────────────────────────────────-┼-────────-┼-────────-┤
│ _initialize()                                                                                                    │ 100.00 % │ 100.00 % │
│ throwInCatch(throwInTry,throwInCatch,enterFinally)                                                               │ 100.00 % │  83.33 % │
│ try02(throwException,enterCatch,enterFinally)                                                                    │ 100.00 % │ 100.00 % │
│ tryecpointCast(useInvalidECpoint,enterCatch,enterFinally)                                                        │  85.71 % │  62.50 % │
│ tryinvalidByteArray2UInt160(useInvalidECpoint,enterCatch,enterFinally)                                           │  85.71 % │  62.50 % │
│ tryinvalidByteArray2UInt256(useInvalidECpoint,enterCatch,enterFinally)                                           │  83.93 % │  56.25 % │
│ try01(throwException,enterCatch,enterFinally)                                                                    │  82.61 % │  58.33 % │
│ try03(throwException,enterCatch,enterFinally)                                                                    │  80.43 % │  58.33 % │
│ tryFinally(throwException,enterFinally)                                                                          │  78.95 % │  60.00 % │
│ tryCatch(throwException,enterCatch)                                                                              │  76.92 % │  50.00 % │
│ tryFinallyAndRethrow(throwException,enterFinally)                                                                │  76.32 % │  60.00 % │
│ tryNest(throwInTry,throwInCatch,throwInFinally,enterOuterCatch)                                                  │  74.67 % │  55.00 % │
│ tryNULL2Ecpoint_1(setToNull,enterCatch,enterFinally)                                                             │  72.53 % │  50.00 % │
│ tryNULL2Uint160_1(setToNull,enterCatch,enterFinally)                                                             │  72.53 % │  50.00 % │
│ tryNULL2Uint256_1(setToNull,enterCatch,enterFinally)                                                             │  72.53 % │  50.00 % │
│ tryNULL2Bytestring_1(setToNull,enterCatch,enterFinally)                                                          │  70.37 % │  50.00 % │
│ tryWithTwoFinally(throwInInner,throwInOuter,enterInnerCatch,enterOuterCatch,enterInnerFinally,enterOuterFinally) │  69.47 % │  47.62 % │
│ tryvalidByteArray2UInt160(enterCatch,enterFinally)                                                               │  67.31 % │  42.86 % │
│ tryvalidByteString2Ecpoint(enterCatch,enterFinally)                                                              │  67.31 % │  42.86 % │
│ throwCall()                                                                                                      │  66.67 % │ 100.00 % │
│ tryUncatchableException(throwException,enterCatch,enterFinally)                                                  │  64.44 % │  50.00 % │
│ tryvalidByteArray2UInt256(enterCatch,enterFinally)                                                               │  63.64 % │  40.00 % │
└-────────────────────────────────────────────────────────────────────────────────────────────────────────────────-┴-────────-┴-────────-┘

**Comment by Jim8y** [comment]
before this pr:


Contract_TryCatch [0xcc347763e4fb1b19f96650f86cf6a031aa7a05ec] [64.64 % - 30.91 %]
┌-─────────────────────────────-┬-────────-┬-────────-┐
│ Method                        │   Line   │   Branch │
├-─────────────────────────────-┼-────────-┼-────────-┤
│ _initialize()                 │ 100.00 % │ 100.00 % │
│ tryecpointCast()              │  79.17 % │  40.00 % │
│ tryinvalidByteArray2UInt160() │  79.17 % │  40.00 % │
│ tryinvalidByteArray2UInt256() │  79.17 % │  40.00 % │
│ codeAfterFinally()            │  77.78 % │ 100.00 % │
│ try02()                       │  77.50 % │  33.33 % │
│ try03()                       │  75.00 % │  33.33 % │
│ tryFinally()                  │  75.00 % │  33.33 % │
│ tryNULL2Ecpoint_1()           │  72.94 % │  38.89 % │
│ tryNULL2Uint160_1()           │  72.94 % │  38.89 % │
│ tryNULL2Uint256_1()           │  72.94 % │  38.89 % │
│ tryCatch()                    │  71.43 % │  33.33 % │
│ throwInCatch()                │  70.97 % │ 100.00 % │
│ tryNULL2Bytestring_1()        │  70.67 % │  35.71 % │
│ tryvalidByteArray2UInt160()   │  68.75 % │  40.00 % │
│ tryvalidByteString2Ecpoint()  │  68.75 % │  40.00 % │
│ throwcall()                   │  66.67 % │ 100.00 % │
│ tryvalidByteArray2UInt256()   │  65.00 % │  33.33 % │
│ try01()                       │  63.16 % │  33.33 % │
│ tryFinallyAndRethrow()        │  61.76 % │  33.33 % │
│ tryNest()                     │  49.25 % │  16.67 % │
│ tryWithTwoFinally()           │  40.87 % │  20.00 % │
│ tryUncatchableException()     │  23.08 % │   0.00 % │
└-─────────────────────────────-┴-────────-┴-────────-┘

**Review by shargon — APPROVED** [review]

**Pull request merged**